### PR TITLE
Tooling Updates

### PR DESCRIPTION
**Pull request by ghickman**
A selection of tooling updates:
* cache dependency downloads in CI (turn on pip caching basically)
* use pyproject.toml as the spec file in `just upgrade <env>`
* cut down .gitignore to not be overselective
* pin Python to 3.7.12 using `.python-version` (_sigh_) so dependabot can work

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ jobs:
       - uses: "actions/setup-python@v2"
         with:
           python-version: "3.7"
+          cache: "pip"
+          cache-dependency-path: requirements.*.txt
```
Comment (benbc) [inline .github/workflows/main.yml:17]: Cool!

File: .python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.7.12
```
Comment (benbc) [inline .python-version:1]: Why is this necessary? We don't use 3.7 for other repos that use dependabot, do we?

File: justfile [hunk 3]
```diff
@@ -83,7 +83,7 @@ upgrade env package="": virtualenv
     #!/usr/bin/env bash
```
Comment (benbc) [inline justfile:83]: I found the comment above here confusing. Github won't let me comment on it directly or suggest a change because it's not part of the diff. It reads as though "all by default" applies to the dev/prod distinction rather than the packages, but would be fine if it read "specify package to upgrade single package, all by default". Would you mind changing it while you're here?

File: .python-version (hunk 2 above)
Comment (ghickman) [reply .python-version:1]: We do!  The docs repo is 3.7 only because that's the latest CloudFlare Pages supports.  Dependabot appears to run with either 3.9 or 3.10 and this is the only way to tell it otherwise.  Since this plugin needs to run in the docs checkout on CloudFlare it's also tied to 3.7.

File: justfile (hunk 3 above)
Comment (ghickman) [reply justfile:83]: Oh yes I see what you mean, fixed 👍

File: .python-version (hunk 2 above)
Comment (benbc) [reply .python-version:1]: I think I understand now. In which case I think the commit message is misleading. I read it as "pin to 3.7 because dependabot needs that version", but I think it's more like "tell dependabot what Python version to use".

Comment (ghickman) [reply .python-version:1]: Good point, message updated 👍

**Review by benbc — APPROVED** [review]
